protocols/kad/src/behaviour.rs

```diff
@@ -2054,8 +2109,17 @@ pub struct PeerRecord {
 /// See [`NetworkBehaviour::poll`].
 #[derive(Debug)]
 pub enum KademliaEvent {
-    /// A query has produced a result.
-    QueryResult {
+    /// An inbound request has been received and handled.
+    //
+    // Note on the difference between 'request' and 'query': A request is a
+    // single request-response style exchange with a single remote peer. A query
+    // is made of multiple requests across multiple remote peers.
+    InboundRequest {
+        request: InboundRequest,
+    },
+
+    /// An outbound query has produced a result.
+    OutboundQueryResult {
```

Comment:
This one could be named `OutboundQueryCompleted` or `OutboundQueryFinished` in the same spirit.